### PR TITLE
Fix/8175 show tooltip instead of deposit guesstimate

### LINKED ISSUE #8175: Don't show an estimated deposit (payout) date in WooPayments UI
### Describe the bug
Currently we show an estimate for when the next deposit will be initiated on the `Payments > Overview` screen:

Screenshot 2024-02-12 at 12 59 21 PM

This date is an estimate for when the deposit will be initiated, based on the merchant's deposit schedule, and our understanding of how payouts are handled in the platform backend (e.g. Stripe).  

There are some limitations and challenges with this estimate:

- The $ will land in the merchant's account some time after the deposit is _initiated_, 1-3 days or maybe more.
- Deposits are initiated in UTC, so our estimate is based on UTC, but merchants (and their banks) are operating in other timezones.
- There are edge cases such as bank holidays that might mean our estimate is incorrect or misleading.

The goal of this issue is to remove the estimated deposit date from the UI (and remove associated code to estimate the date). Instead of showing the date we'll show a help icon or `Learn more` link, empowering new merchants with information so they understand when they will get paid.

### To Reproduce
View `Payments > Overview`.

### Actual behavior
- Estimated payout date shown, which may not be 100% accurate at all times, and doesn't provide any context on how payouts work.

### Expected behavior
- Show the merchant a summary of their deposit schedule. 
- Provide info (in docs or a tooltip) about how deposits work so merchants can understand how & when they will receive funds.

### Additional context
See also related issues:

- https://github.com/Automattic/woocommerce-payments/issues/8026
- Automattic/woocommerce-payments-server#4435

### PR DESCRIPTION
**Pull request by haszari**
Fixes #8175 

Screenshot 2024-02-14 at 12 42 39 PM

Tooltip text is as follows:

> Deposits are initiated based on the following criteria:
> - The [pending period](https://woo.com/document/woopayments/deposits/deposit-schedule/#pending-period-chart) in your country
> - Your account's [available funds](https://woo.com/document/woopayments/deposits/deposit-schedule/#available-funds)
> - Your [deposit schedule](https://woo.com/document/woopayments/deposits/change-deposit-schedule/) settings

#### Changes proposed in this Pull Request
- Removes the estimated next deposit date from deposit overview screen (and related code)
- Adds a tooltip with more info and links to docs about what factors affect deposits / payouts

#### Testing instructions

* Get your store in a state where you can expect payouts (e.g. outside of new account waiting period)
* View `Payments > Overview`
* Check the deposit box payout schedule info makes sense, and the tooltip is helpful.
* Test different payout schedules – the format string(s) for the schedule summary changed in this PR (to remove the estimated date part).

Bonus points:

- Test other store states to ensure the copy & tooltip are relevant and helpful. (Should be consistent/unchanged from previous logic)

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : qa testing not required
- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable. no changes to critical flows
- [x] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable. TBD - will need adding to What's new post, with link to relevant P2 discussion

## more screenshots

With a monthly schedule:

Screenshot 2024-02-15 at 9 54 26 AM

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8184` or branch name `fix/8175-show-tooltip-instead-of-deposit-guesstimate` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/8175-show-tooltip-instead-of-deposit-guesstimate) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: bb08bd3ec468186309af6c87c29b582342b5539a
- Build time: 2024-02-18 20:47:55 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/components/deposits-overview/index.tsx [hunk 1]
```diff
@@ -119,6 +193,12 @@ const DepositsOverview: React.FC = () => {
 						depositsSchedule={ account.deposits_schedule }
 						showNextDepositDate={ availableFunds > 0 }
 					/>
+					<ClickTooltip
+						content={ nextDepositHelpContent }
```
Comment (haszari) [inline client/components/deposits-overview/index.tsx:196]: I really wanted to put the tooltip markup in `children`, [composition](https://felixgerschau.com/react-component-composition/) style:

```JSX
<ClickTooltip>
  Deposits are initiated based on the following criteria:

  <ul>
    <li>The pending period in your country</li>
    <li>(etc)</li>
  </ul>
</ClickTooltip>
```

However, our tooltip uses `children` for something else – I think it allows the client to use a different element for the button/element that user clicks to _show_ the tooltip. 

Would love to refactor our `ToolTip` and friends to support arbitrary tooltip content as `children` … though not important to do now!

**Comment by haszari** [comment]
> `Deposit Schedule renders › with a monthly schedule on the 2nd`

Looks like I have some tests to fix 📓

**Comment by haszari** [comment]
>  ● Deposits Overview information › Component Renders

I also need to update snapshots.

**Comment by haszari** [comment]
Also tested with account soft blocked by platform, looking good:

Screenshot 2024-02-15 at 10 56 52 AM

File: client/components/deposits-overview/index.tsx (hunk 1 above)
Comment (Jinksi) [reply client/components/deposits-overview/index.tsx:196]: I agree!

💭 I think this component was originally based on `@wordpress/components` [tooltip](https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/tooltip#children-reactreactelement), for which `children` is used for "The element to which the tooltip should anchor.".

For `@wordpress/components` [popover](https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/popover#children-reactnode) component, however, `children` is used for "elements rendered as the popover's content." 👍

File: client/components/deposits-overview/index.tsx [hunk 2]
```diff
@@ -119,6 +193,12 @@ const DepositsOverview: React.FC = () => {
...
+						className="wcpay-deposits-overview__schedule__tooltip"
+						buttonIcon={ <HelpOutlineIcon /> }
+						buttonLabel={ 'Deposit schedule tooltip' }
+					/>
```
Comment (Jinksi) [inline client/components/deposits-overview/index.tsx]: ⛏️ this `className` may be redundant, AFAICT it isn't references anywhere.

**Review by Jinksi — APPROVED** [review]
This tests well @haszari 🙌

I've added some commits with minor code changes in 198303a6a0022601dc44b545a2288c57a93fb23f, c3bd550cf49c00db4802b495f095d60797772d89 and 7de0b054480bca00a96397c42ce093dbf6100aa8.

I've tested the message against daily, weekly, monthly deposit schedules. ✅
The tooltip renders well and is navigable via keyboard. ✅

⚠️ Note that when the deposit schedule is set to manual via Stripe Dashboard, the tooltip is rendered without a schedule message.

image

I believe this is a rare edge case, but we could bail on rendering the entire schedule message section if check if you think it's worth avoiding for manual scheduled.


💡 A future enhancement may involve adding this tooltip to the Deposits screen, related to #8026.

Screenshot 2024-02-15 at 12 00 55

**Comment by haszari** [comment]
> I've added some commits with minor code changes in https://github.com/Automattic/woocommerce-payments/commit/198303a6a0022601dc44b545a2288c57a93fb23f, https://github.com/Automattic/woocommerce-payments/commit/c3bd550cf49c00db4802b495f095d60797772d89 and https://github.com/Automattic/woocommerce-payments/commit/7de0b054480bca00a96397c42ce093dbf6100aa8.

Thanks @Jinksi  for catching these tidy ups!

> ⚠️ Note that when the deposit schedule is set to manual via Stripe Dashboard, the tooltip is rendered without a schedule message.
I believe this is a rare edge case, but we could bail on rendering the entire schedule message section if check if you think it's worth avoiding for manual scheduled.

I like this idea – though pulled back when I realised I need to add the same logic in `DepositSchedule` to the outer card component (around the tooltip). And the logic is a little complex, need to check the schedule `interval` is one of 3 known strings. So … leaving as is.

I'm thinking when we adapt this for #8026 we can move the tooltip inside `DepositSchedule`.

File: client/components/deposits-overview/index.tsx (hunk 2 above)
Comment (haszari) [reply client/components/deposits-overview/index.tsx]: Removed! adfc76dee01a5b9f57937de92e5bf0b38fe1191a

**Comment by haszari** [comment]
> ❌ E2E Tests - Pull Request / WC - latest | wcpay - merchant
> ❌ E2E Tests - Pull Request / WC - latest | subscriptions - merchant

Looks like there's a bunch of unrelated failures breaking the tests. Unrelated to this PR so merging.

> Failed to load resource: the server responded with a status of 400 (Bad Request)
>       at Page.<anonymous> (../../../../../Users/ronrennick/Sites/solaris/woocommerce/tests/e2e/env/src/setup/jest.setup.js:164:3)
      at Page._onLogEntryAdded (node_modules/puppeteer/lib/Page.js:224:12)

> Order > Full refund › should be able to view a refunded transaction
>    TimeoutError: Element li.woocommerce-timeline-item (text: "A payment of $0.00 was successfully refunded.") not found

> Order > Status Change › Change Status of order to Cancelled › When Order Status changed to Cancel, show Cancel Confirmation modal, change status to Cancel if confirmed
>    : Timeout - Async callback was not invoked within the 100000 ms timeout specified by jest.setTimeout.Timeout - Async callback was not invoked within the 100000 ms timeout specified by jest.setTimeout.Error:

These are possibly due to recent release of Woo core 8.6?

**Pull request merged**